lib/AST/ClangTypeConverter.h

```diff
@@ -79,8 +79,19 @@ class ClangTypeConverter :
   /// Swift declaration.
   Decl *getSwiftDeclForExportedClangDecl(const clang::Decl *decl) const;
 
+  /// Translate Swift generic arguments to template arguments.
+  ///
+  /// \returns an error if any parameter in \param templateParams is a non-type
+  /// template parameter or if there is a conversion failure for one of the
+  /// generic arguments.
```

Comment:
Sorry for bringing this up again, but this isn't accurate. As you said, the Clang importing ensures that `templateParams` does not have any non-type template parameters. You could shorten this to:

```
\returns true on success, false if there is a conversion failure for a generic argument
```

Otherwise, it isn't clear whether `true` stands for success or for failure (different parts of the compiler use different conventions).

Reply:
Right, thanks. That's a good point. Will fix.